macosx/Controller.mm

```diff
@@ -5246,6 +5246,22 @@ - (void)setWindowSizeToFit
     }
 }
 
+- (CGFloat)calculateScrollViewHeightWithDockAdjustment
+{
+    CGFloat height = self.scrollViewHeight;
+
+    // Get the main screen's visible frame
+    NSScreen* screen = self.fWindow.screen;
+    if (screen)
+    {
+        // This frame respects the Dock and menu bar
+        NSRect visibleFrame = screen.visibleFrame;
+        height = MIN(height, visibleFrame.size.height - [self toolbarHeight] - [self mainWindowComponentHeight]);
```

Comment:
btw, why don't we adjust window size itself?
This look very error prone, if we want to change/reorder layout in any way.

Reply:
Something like this?

```
{
    CGFloat height = self.scrollViewHeight;

    // Get the main screen's visible frame
    NSScreen* screen = self.fWindow.screen;
    if (screen)
    {
        // This frame respects the Dock and menu bar
        NSRect visibleFrame = screen.visibleFrame;
        height = MIN(height, visibleFrame.size.height - [self toolbarHeight] - [self mainWindowComponentHeight]);

        // Adjust the window size
        NSRect newFrame = [self.window frame];
        newFrame.size.height = height;
        [self.window setFrame:newFrame display:YES animate:YES];
    }

    return height;
}

```

Reply:
NVM, I see now this calculations are for scroll view constraint.

What I suggest is to set scroll view bottom constraint to always follow window bottom and then just adjust window size to be in visible frame.

This would be a separate refactoring, your fix will do for now.